pkg/sql/sql_cursor.go

```diff
@@ -45,6 +42,9 @@ func (p *planner) DeclareCursor(ctx context.Context, s *tree.DeclareCursor) (pla
 		name: s.String(),
 		constructor: func(ctx context.Context, p *planner) (_ planNode, _ error) {
 			if p.extendedEvalCtx.TxnImplicit {
+				if s.Hold {
```

Comment:
can this no-op? or do we have an internal limitation that cursors require txns?

```
otan=# declare a cursor with hold for select 1;
DECLARE CURSOR
```

also wonder if we should have separate issues numbers for this and the closeAll scenario below for telemetry purposes

Reply:
by default, cursors require transactions (in PG too). `with hold` is special because it does not require a transaction, but that is the thing we don't support. no-oping would be confusing IMO, since the user may think that it was created successfully.

Reply:
i'm fine with one issue personally, because the underlying limitation for both is "cannot use a WITH HOLD cursor outside of an explicit transaction. both places where we return the `unimplemented` error will be fixed at the same time.